### PR TITLE
[Stable] Fix some neq predicate issues

### PR DESCRIPTION
**Pull request by kasper-piskorski**

### CODE REVIEW

File: grakn-graql/src/test/java/ai/grakn/graql/internal/reasoner/ReasoningTests.java [hunk 1]
```diff
@@ -655,6 +655,76 @@ public void transRelationWithNeqPredicate(){
         assertTrue(answers2.containsAll(answers));
     }
 
+    @Test
+    public void transConjunctionWithNeqPredicate(){
```
Comment (fppt) [inline grakn-graql/src/test/java/ai/grakn/graql/internal/reasoner/ReasoningTests.java]: Could you please provide more detailed test names or comment what these tests actually check?

File: grakn-graql/src/test/java/ai/grakn/graql/internal/reasoner/ReasoningTests.java [hunk 2]
```diff
@@ -655,15 +662,32 @@ public void transRelationWithNeqPredicate(){
         assertTrue(answers2.containsAll(answers));
     }
 
+    /**
+     * Tests a scenario in which the neq predicate binds free variables of two recursive equivalent relations.
```
Comment (fppt) [inline grakn-graql/src/test/java/ai/grakn/graql/internal/reasoner/ReasoningTests.java]: 👍

**Comment by fppt** [comment]
@kasper-piskorski please merge this into master

**Comment by kasper-piskorski** [comment]
merged here: https://github.com/graknlabs/grakn/pull/2052

**Pull request merged**